helpers/events.js

```diff
@@ -383,3 +385,77 @@ exports.removeEventsByContractStatus = async (contract) => {
   const correspondingSubsIds = correspondingSubs.map(sub => sub.sub._id);
   await Event.deleteMany({ startDate: { $gt: contract.endDate }, subscription: { $in: correspondingSubsIds }, isBilled: false });
 };
+
+function getFullTitleFromIdentity(identity) {
+  const lastname = _.get(identity, 'lastname') || '';
+  let fullTitle = [
+    _.get(identity, 'title'),
+    _.get(identity, 'firstname'),
+    lastname.toUpperCase(),
+  ];
+
+  fullTitle = fullTitle.filter(value => !_.isEmpty(value));
+
+  return fullTitle.join(' ');
+}
+
+exports.exportWorkingEventsHistory = async (startDate, endDate) => {
+  const searchStartDate = moment(startDate).startOf('day').toDate();
+  const searchEndDate = moment(endDate).endOf('day').toDate();
+
+  const query = {
+    type: { $in: [INTERVENTION, INTERNAL_HOUR] },
+    $or: [
+      { startDate: { $lte: searchEndDate, $gte: searchStartDate } },
+      { endDate: { $lte: searchEndDate, $gte: searchStartDate } },
+      { endDate: { $gte: searchEndDate }, startDate: { $lte: searchStartDate } },
+    ],
+  };
+
+  const events = await Event.find(query)
+    .sort({ startDate: 'desc' })
+    .populate({ path: 'auxiliary', select: 'identity' })
+    .populate({ path: 'customer', select: 'identity' })
+    .populate({ path: 'sector' })
+    .lean();
+
+  const header = [
+    'Type',
+    'Début',
+    'Fin',
+    'Répétition',
+    'Secteur',
+    'Auxiliaire',
+    'Bénéficiaire',
+    'Divers',
+    'Facturé',
+    'Annulé',
+    'Status de l\'annulation',
+    'Raison de l\'annulation',
+  ];
+
+  const rows = events.map((event) => {
```

Comment:
par coherence dans toute l'appli on utilise plutot des boucles for de type `for ( const ... of ....)`
et tu push a chaque iteration dans ton csv global

Reply:
Pourtant `map` est parfaitement adapté à ce genre de cas, pourquoi s'en passer?

Reply:
D'acc